### ISSUE TITLE
Hook for RadiusDamage

### ISSUE THREAD

**Issue by thegrb93** [Facepunch/garrysmod-requests#784]
I want to make my own prop explosion shielding because the current one has some problems, and I want to scale the damage rather than block it completely. I looked everywhere in sourcesdk and couldn't find explosion shielding implemented anywhere (except for the world blocking explosions) so I'm pretty sure there's a custom one that garrysmod uses. Would it possible to get that control with this hook?

Or, if you want to fix the current shielding code.. The problem is that there's only one trace done, so if the explosion is tracing a player and hits a small prop first, it will damage the player even if a big prop is behind the small prop.

**Comment by LuaTenshi** [comment]
You could use the [EntityTakeDamage](https://wiki.garrysmod.com/page/GM/EntityTakeDamage) hook. Find if the damage [IsExplosionDamage](https://wiki.garrysmod.com/page/CTakeDamageInfo/IsExplosionDamage) and scale it using [ScaleDamage](https://wiki.garrysmod.com/page/CTakeDamageInfo/ScaleDamage). 
From there you could also use as many traces as you need to find any "shields" that may be in place etc.

**The above is a response on how you may be able to make your own shielding.**
Also I am sure you will be able to find things like radius and original damage using [GetKeyValues](https://wiki.garrysmod.com/page/Entity/GetKeyValues)

**Comment by thegrb93** [comment]
The player doesn't take any damage if the explosion is shielded. I was thinking I could recheck the radius of an explosion if it hits anything, but I don't think there's a way to get the radius of the explosion. How would you get the radius of an explosion created from util.BlastDamage?

**Comment by Kefta** [comment]
I believe BlastDamage is a rename of RadiusDamage if you want to port that into Lua and handle it yourself.

**Comment by thegrb93** [comment]
Yeah, I know about RadiusDamage, but how would I replace it so explosions use mine instead of Garrysmod's?

**Comment by LuaTenshi** [comment]
Haven't tested it but have you tried using [BlastDamageInfo](https://wiki.garrysmod.com/page/util/BlastDamageInfo) and while creating the explosion on the damage info you would add in your radius... 
> d.DamageRadius = 300
I think that may just get passed along inside the [CTakeDamageInfo](https://wiki.garrysmod.com/page/Category:CTakeDamageInfo)

**Comment by thegrb93** [comment]
That doesn't help for other addons' or the engine's explosions.

**Comment by thegrb93** [comment]
What about a hook for RadiusDamage. To override explosions of props, rpgs, anything we want.
